core/kernel/tee_ta_manager.c

```diff
@@ -982,7 +982,14 @@ TEE_Result tee_ta_instance_stats(void *buf, uint32_t *buf_size)
 			ta_count++;
 
 	sz = sizeof(struct tee_ta_dump_stats) * ta_count;
-	if (!buf || *buf_size < sz) {
+	if (sz == 0) {
+		/*
+		 * sz = 0 means there is no UTA, set buf_size
+		 * as 0 and return no item found.
+		 */
+		*buf_size = 0;
```

Comment:
Updating `buf_size` is not necessary I believe. It is supposed to be used only when return value is `TEE_SUCCESS` or `TEE_ERROR_SHORT_BUFFER`.

Reply:
done. thanks for the reviewing.